### PR TITLE
Scripts for remote centos experiment environment deployment 

### PR DESCRIPTION
**Pull request by Dual2Core**
Fixes issue remote CentOS7 experiment environment deployment.

Summary of changes:
- prepare_host_env.sh - for building swan locally
- centos_deploy_playbook.yml - ansible playbook used to prepare remote nodes for swan experiments

Testing done:
- Installed on 3 VM's and successfully runned memcached sensitivity profile experiment from quickstart.

### CODE REVIEW

**Comment by Dual2Core** [comment]
run integration tests

**Comment by mpatelcz** [comment]
run integration tests

File: ansible/centos_deploy_playbook.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      copy:
+        src: ~/go/src/github.com/intelsdi-x/swan/build/
+        dest: /opt/swan
+    - name: Copy Swan services
+      copy:
+        src: ~/go/src/github.com/intelsdi-x/swan/vagrant/
+        dest: /swan_services
+    - name: Experiment environment provisioning script run
+      environment:
+        SWAN_USER: centos
```
Comment (squall0gd) [inline ansible/centos_deploy_playbook.yml]: This should be configurable with global_vars

File: ansible/centos_deploy_playbook.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        HOME_DIR: /home/centos/
```
Comment (squall0gd) [inline ansible/centos_deploy_playbook.yml]: same here

File: ansible/install_mutilate_and_config.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Install mutilate
+git clone https://github.com/leverich/mutilate
+cd mutilate
+scons
+sudo ln -sf `pwd`/mutilate /usr/bin/
+
+# Configure system for swan
+#echo 0 > /proc/sys/net/ipv4/tcp_syncookies
+#ulimit -n 10000
```
Comment (squall0gd) [inline ansible/install_mutilate_and_config.sh]: If it's not used - remove it

File: ansible/install_exp_env_centos.sh [hunk 4]
```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env bash
```
Comment (squall0gd) [inline ansible/install_exp_env_centos.sh]: Why not just a symbolic link to https://github.com/intelsdi-x/swan/blob/master/vagrant/provision_experiment_environment.sh ?

File: ansible/install_mutilate_and_config.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Install mutilate
```
Comment (squall0gd) [inline ansible/install_mutilate_and_config.sh]: Missing shellbang

File: ansible/prepare_host_env.sh [hunk 6]
```diff
@@ -0,0 +1,36 @@
+if [ "$EUID" -ne 0 ]
```
Comment (squall0gd) [inline ansible/prepare_host_env.sh]: Missing shellbang; This file isn't used anywhere

File: ansible/prepare_host_env.sh [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	then echo "Please run this setup with 'source' command"
+	exit
+fi
+
+
+### Install required packages
+echo "Installing required packages..."
+
+# git
+apt install git -y
```
Comment (squall0gd) [inline ansible/prepare_host_env.sh]: Swan is not supporting `Debian`'s

File: ansible/local_scripts/goinstall.sh [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+VERSION="1.8.1"
+
+print_help() {
+    echo "Usage: bash goinstall.sh OPTION"
+    echo -e "\nOPTIONS:"
+    echo -e "  --32\t\tInstall 32-bit version"
+    echo -e "  --64\t\tInstall 64-bit version"
+    echo -e "  --arm\t\tInstall armv6 version"
+    echo -e "  --remove\tTo remove currently installed version"
```
Comment (squall0gd) [inline ansible/local_scripts/goinstall.sh]: You don't need any other options than `64`, b/c Swan is supporting 64 bit platforms only.

File: ansible/prepare_host_env.sh [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+# golang
+bash local_scripts/goinstall.sh --64
+source ~/.bashrc
+
+# caffe
+
+
+### Install and configure caffe
+
```
Comment (squall0gd) [inline ansible/prepare_host_env.sh]: empty comments in L22 and L25

File: ansible/install_exp_env_centos.sh (hunk 4 above)
Comment (Dual2Core) [reply ansible/install_exp_env_centos.sh]: Because of changed swan services paths.

File: ansible/prepare_host_env.sh (hunk 7 above)
Comment (Dual2Core) [reply ansible/prepare_host_env.sh]: Yet it can be build on it.

File: ansible/prepare_host_env.sh (hunk 6 above)
Comment (Dual2Core) [reply ansible/prepare_host_env.sh]: It is used manually. I'll create readme for that.

File: ansible/install_exp_env_centos.sh (hunk 4 above)
Comment (iwankgb) [reply ansible/install_exp_env_centos.sh]: @Dual2Core I don't think it's a good idea. You copied *whole* script while you could *parametrise* existing one and use it.

File: ansible/local_scripts/goinstall.sh [hunk 10]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+set -e
+
+VERSION="1.8.1"
```
Comment (iwankgb) [inline ansible/local_scripts/goinstall.sh]: @Dual2Core - upgrade to 1.9.

File: ansible/local_scripts/goinstall.sh [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+print_help() {
+    echo "Usage: bash goinstall.sh OPTION"
+    echo -e "\nOPTIONS:"
+    echo -e "  --32\t\tInstall 32-bit version"
```
Comment (iwankgb) [inline ansible/local_scripts/goinstall.sh]: @Dual2Core - are all these options required? If not - [delete them](https://martinfowler.com/bliki/Yagni.html), please.

File: ansible/local_scripts/goinstall.sh [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+echo "Extracting ..."
+tar -C "$HOME" -xzf /tmp/go.tar.gz
+mv "$HOME/go" "$HOME/.go"
+touch "$HOME/.bashrc"
+{
+    echo '# GoLang'
+    echo 'export GOROOT=$HOME/.go'
+    echo 'export PATH=$PATH:$GOROOT/bin'
+    echo 'export GOPATH=$HOME/go'
+    echo 'export PATH=$PATH:$GOPATH/bin'
```
Comment (iwankgb) [inline ansible/local_scripts/goinstall.sh]: @Dual2Core - you declare `PATH` twice.

File: ansible/local_scripts/goinstall.sh [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+fi
+echo "Extracting ..."
+tar -C "$HOME" -xzf /tmp/go.tar.gz
+mv "$HOME/go" "$HOME/.go"
+touch "$HOME/.bashrc"
+{
+    echo '# GoLang'
+    echo 'export GOROOT=$HOME/.go'
+    echo 'export PATH=$PATH:$GOROOT/bin'
+    echo 'export GOPATH=$HOME/go'
```
Comment (iwankgb) [inline ansible/local_scripts/goinstall.sh]: @Dual2Core I don't think you need `GOPATH`.

File: ansible/local_scripts/goinstall.sh (hunk 10 above)
Comment (iwankgb) [reply ansible/local_scripts/goinstall.sh]: (and consider using another of existing provisioning scripts)

**Comment by Dual2Core** [comment]
run integration tests

File: ansible/install_exp_env_centos.sh (hunk 4 above)
Comment (squall0gd) [reply ansible/install_exp_env_centos.sh]: So parametrize it :P

File: ansible/prepare_host_env.sh (hunk 7 above)
Comment (squall0gd) [reply ansible/prepare_host_env.sh]: It can be built b/c golang doesn't care about it. Swan in it's basics doesn't support Debians ;P

Comment (iwankgb) [reply ansible/prepare_host_env.sh]: As far as I understand this script is to be run on developer's machine. OS is not very relevant then.

File: ansible/prepare_host_env.sh [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+rm -rf "$HOME/.go"
+
+# Save Go to .go
+mv "$HOME/go" "$HOME/.go"
+touch "$HOME/.bashrc"
+{
+	echo 'export GOROOT=$HOME/.go'
+	echo 'export PATH=$PATH:$GOROOT/bin'
+	echo 'export GOPATH=$HOME/go'
+	echo 'export PATH=$PATH:$GOPATH/bin'
```
Comment (iwankgb) [inline ansible/prepare_host_env.sh]: @Dual2Core - declare `PATH` only once.

File: ansible/prepare_host_env.sh [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+# Be sure to uninstall old go
+rm -rf "$HOME/.go"
+
+# Save Go to .go
+mv "$HOME/go" "$HOME/.go"
+touch "$HOME/.bashrc"
+{
+	echo 'export GOROOT=$HOME/.go'
+	echo 'export PATH=$PATH:$GOROOT/bin'
+	echo 'export GOPATH=$HOME/go'
```
Comment (iwankgb) [inline ansible/prepare_host_env.sh]: @Dual2Core remove, this is default value for golang >= 1.8, `glide` might need to be updated. See PR: https://github.com/Masterminds/glide/issues/771

File: ansible/prepare_host_env.sh [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+	echo 'export PATH=$PATH:$GOPATH/bin'
+} >> "$HOME/.bashrc"
+
+export GOROOT=$HOME/.go
+export PATH=$PATH:$GOROOT/bin
+export GOPATH=$HOME/go
+export PATH=$PATH:$GOPATH/bin
```
Comment (iwankgb) [inline ansible/prepare_host_env.sh]: @Dual2Core see my comments regarding environmental variable above.

File: ansible/prepare_host_env.sh [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+
+mkdir -p $HOME/go/{src,pkg,bin}
+echo -e "\nGo $VERSION was installed.\n"
+rm -f /tmp/go.tar.gz
+
+### Get latest Swan repository
+git clone https://github.com/intelsdi-x/swan.git
```
Comment (iwankgb) [inline ansible/prepare_host_env.sh]: @Dual2Core use `go get` instead.

File: ansible/prepare_host_env.sh [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+
+### Build Swan
+mkdir -p $GOPATH/src/github.com/intelsdi-x/swan
+mv swan $GOPATH/src/github.com/intelsdi-x/
```
Comment (iwankgb) [inline ansible/prepare_host_env.sh]: @Dual2Core remove after switching to `go get`.

File: ansible/README.md [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+
+# Ansible swan deployment playbook
+
+## Introduction
+To deploy your own swan experiments to remote nodes we provide a environment preparation scripts both for your host and remote machines. Note that currently host preparation script uses `apt` package manager so `Debian` or `Ubuntu` would be a great choice. Remote installation scripts(ansible) on the other hand, are written for `CentOS7`. 
+
+## Usage
+First, to prepare your own machine for development and build Swan locally run following:
+```bash
+git clone https://github.com/intelsdi-x/swan.git
```
Comment (iwankgb) [inline ansible/README.md]: @Dual2Core use `go get` instead. Cloning golang source code to location outside of `GOPATH` is not a good idea.

File: ansible/install_exp_env_centos.sh (hunk 4 above)
Comment (iwankgb) [reply ansible/install_exp_env_centos.sh]: Or just symlink Swan's `vagrant` directory to `/vagrant`.

File: ansible/README.md [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+# Ansible swan deployment playbook
+
+## Introduction
+To deploy your own swan experiments to remote nodes we provide a environment preparation scripts both for your host and remote machines. Note that currently host preparation script uses `apt` package manager so `Debian` or `Ubuntu` would be a great choice. Remote installation scripts(ansible) on the other hand, are written for `CentOS7`.
```
Comment (iwankgb) [inline ansible/README.md]: @Dual2Core use capital S when you write about Swan, please.

File: ansible/README.md (hunk 19 above)
Comment (Dual2Core) [reply ansible/README.md]: I get your idea, but `prepare_host_env.sh` assumes that there is no Golang installed, so I shouldn't tell to use it to download files to install Golang. That would be strange. So I can leave it like it is in the tutorial or remove whole Go installation part and leave a note in requirements section to have go installed before running the script. However, I can also add `rm -rf swan` and then Swan repo will exist only at Gopath location.

**Comment by Dual2Core** [comment]
run integration tests

**Comment by Dual2Core** [comment]
run integration tests

File: ansible/README.md (hunk 19 above)
Comment (iwankgb) [reply ansible/README.md]: If we download the project source just in order to get hold of Ansible manifests and shell scripts then cloning is good solution.

Comment (iwankgb) [reply ansible/README.md]: @Dual2Core I do like your idea about replacing `git clone` with downloading single shell script. Implement it, please. 👍

File: vagrant/provision_experiment_environment.sh [hunk 21]
```diff
@@ -132,7 +133,8 @@ systemctl restart cassandra
 daemonStatus cassandra
 
 echo "----------------------------- Install InfluxDB (`date`)"
-cp /vagrant/influxdb/influxdb.service /etc/systemd/system
+#cp /vagrant/influxdb/influxdb.service /etc/systemd/system
```
Comment (iwankgb) [inline vagrant/provision_experiment_environment.sh]: @Dual2Core remove commented out line, please.

**Comment by Dual2Core** [comment]
run integration tests

**Comment by Dual2Core** [comment]
run integration tests

**Comment by iwankgb** [comment]
Due to AWS issues it is not possible to test this PR even locally :/

**Comment by iwankgb** [comment]
It should be possible to run the tests next week.

**Comment by iwankgb** [comment]
I have just had a chat with @gryf about this PR. We believe that it belongs to another project and should not be merged to Swan. I am not going to delete the branch but I'm going to close the PR. Reopen or ping me when necessary.
CC: @Dual2Core, @mpatelcz.

**Pull request closed without merging**